tests/lib/util/glob-util.js

```diff
@@ -221,6 +221,50 @@ describe("globUtil", function() {
             ]);
         });
 
+        it("should silently ignore default ignored files if not passed explicitly", function() {
+            var directory = getFixturePath("glob-util", "hidden");
+            var patterns = [directory];
+            var result = globUtil.listFilesToProcess(patterns, {
+                cwd: getFixturePath()
+            });
+
+            assert.equal(result.length, 0);
+        });
+
+        it("should ignore and warn for default ignored files when passed explicitly", function() {
```

Comment:
@nzakas Is this the expected behavior? I understood otherwise from these [comments](https://github.com/eslint/eslint/issues/4828#issuecomment-194535273)